src/sentry/tsdb/base.py

```diff
@@ -83,6 +83,25 @@ class TSDBModel(Enum):
     # the number of events blocked due to being blacklisted
     key_total_blacklisted = 502
 
+    # the number of events filtered by ip
+    project_total_received_ip_address = 601
+    # the number of events filtered by release
+    project_total_received_release_version = 602
+    # the number of events filtered by error message
+    project_total_received_error_message = 603
+    # the number of events filtered by browser extension
+    project_total_received_browser_extensions = 604
+    # the number of events filtered by legacy browser
+    project_total_received_legacy_browsers = 605
+    # the number of events filtered by localhost
+    project_total_received_localhost = 606
+    # the number of events filtered by web crawlers
+    project_total_received_web_crawlers = 607
+    # the number of events filtered by invalid csp
+    project_total_received_invalid_csp = 608
+    # the number of events filtered by invalid origin
+    project_total_received_cors = 609
```

Comment:
It is what it is for this patch, but we should really find a better pattern for this hierarchical key structure in TSDB 2.

Reply:
I know you're not asking for changes from me on this, but I would be curious for your explanation on why this is the way it is and what makes it bad if you have time at some point to go over it.

Reply:
I can't totally speak to why it's defined the way it is today (since this predates my involvement) but I don't think that any sort of hierarchical or dynamic keys were part of the original design goals, which made sense at the time.

The current implementation limits the ability to do things like have a project received statistic that is also segmented by things like the custom filter ID that we talked about earlier in this review. This also just feels like a lot of repetition. It seems like we'd be better suited to have a `project_total_received` statistic with different filtering reasons such as the statically defined filter types, as well as the dynamic custom filter IDs. We also do lots of hacky stuff with the "internal" key to support dynamic keys due of the inflexibility of the key designs. (I'm not totally sure if this answers your question.)